<launch>

	<arg name="logitech_cam_file" value="$(find estimation)/config/cTw_cam0.yaml"/>
  <arg name="cam_file" value="$(arg logitech_cam_file)" />
  	
  <!-- </aruco_detection_0/camera/rgb -->
  <!-- <remap from="/aruco_detection_0/camera/rgb" to="/mushr/camera/rgb"/> -->
  <!-- <remap from="/mushr/camera/rgb" to="/aruco_detection_0/camera/rgb"/> -->
  <group ns="mushr">
  	
    <node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="aruco_detection_nodelet_t" args="load perception/aruco_detection_nodelet_t standalone_nodelet" output="screen">
        <rosparam command="load" file="$(find perception)/config/aruco_detection.yaml" />
    </node>
    <node pkg="nodelet" type="nodelet" name="aruco_cTw_nodelet_t" args="load estimation/aruco_cTw_nodelet_t standalone_nodelet" output="screen">
      <rosparam command="load" file="$(arg cam_file)" />
    </node>  

    <node pkg="nodelet" type="nodelet" name="aruco_wTc_nodelet_t" args="load estimation/aruco_wTc_nodelet_t standalone_nodelet" output="screen">
      <rosparam command="load" file="$(arg cam_file)" />
      <param name="pose_topic" value="/mushr/aruco_cTw/robot_pose" />
      <param name="image_topic" value="/mushr/camera/rgb"/>
      <param name="goal_pose_topic" value="/mushr/goal_pos"/>
			<param name="goal_rad_topic" value="/mushr/goal_radius"/>
			<param name="trajectory_topic" value="/mushr/ml4kp_traj"/>
			<param name="camera_frame" value="camera"/>
			<param name="world_frame" value="world"/>
			<param name="robot_frame" value="robot_0"/>
    </node>   

    <!-- <node pkg="nodelet" type="nodelet" name="plant_estimator_nodelet_t" args="load estimation/plant_estimator_nodelet_t standalone_nodelet" output="screen">
			<param name="plan_topic" value="/mushr/ml4kp_plan"/>
			<param name="trajectory_topic" value="/mushr/ml4kp_traj"/>
			<param name="simulation_step" value="0.1"/>
			<param name="observation_dim" value="3"/>
			<param name="world_frame" value="world"/>
			<param name="robot_frame" value="robot_0"/>
    </node>    -->
 
   </group>

	<!-- <node pkg="tf2_ros" type="static_transform_publisher" name="tf_world"  -->
		<!-- args="0 0 0 0 0 0 1  marker_50 world" /> -->
  <arg name="world_marker_tran" value="0 0 0" />
  <arg name="world_marker_quat" value="0 0 1 0" />

	<node pkg="tf2_ros" type="static_transform_publisher" name="tf_world" 
		args="$(arg world_marker_tran) $(arg world_marker_quat) marker_54 world" />
</launch>
